modules/solid_mechanics/src/actions/CommonSolidMechanicsAction.C

```diff
@@ -29,8 +31,98 @@ CommonSolidMechanicsAction::CommonSolidMechanicsAction(const InputParameters & p
 void
 CommonSolidMechanicsAction::act()
 {
+
   // check if sub-blocks block are found which will use the common parameters
-  auto action = _awh.getActions<QuasiStaticSolidMechanicsPhysicsBase>();
-  if (action.size() == 0)
+  auto actions = _awh.getActions<QuasiStaticSolidMechanicsPhysicsBase>();
+  if (actions.size() == 0)
     mooseWarning("Common parameters are supplied, but not used in ", parameters().blockLocation());
+
+  // Add disp-variables
+  if (_current_task == "add_variable")
+  {
+
+    // helper variables to collect what the user wants
+    bool do_add_variables = isParamSetByUser("add_variables") && getParam<bool>("add_variables");
+    std::vector<VariableName> displacement_variables;
+    if (isParamSetByUser("displacements"))
+    {
+      const auto v = getParam<std::vector<VariableName>>("displacements");
+      if (v.size() > 0)
+        displacement_variables.insert(displacement_variables.end(), v.begin(), v.end());
+    }
```

Comment:
Isn't this just
```suggestion
    auto displacement_variables = getParam<std::vector<VariableName>>("displacements");
```
?

Reply:
Yes, thank you.